src/SpecFile.cpp

```diff
@@ -4348,8 +4355,11 @@ bool SpecFile::load_file( const std::string &filename,
           triedOrtecLM = false, triedMicroRaider = false, triedAram = false,
           triedTka = false, triedMultiAct = false, triedPhd = false,
           triedLzs = false, triedXmlScanData = false, triedJson = false,
-          tried_gxml = false;
+          tried_gxml = false, triedRadiaCode = false;
       
+      if( orig_file_ending.empty() )
```

Comment:
I had to add this to make autodetection work correctly. Without it, neither cambio nor convert_to_n42 were able to guess the file format.

Reply:
I'll also take a look at this - in practice, filenames often get mangled, or given different extensions, by the time the user gets them, so I'll make sure the code will parse even without a filename.

Reply:
This was the least disruptive way I could think of to get autodetection to work. I just wanted you to be aware of this change so you didn't think it was an unintentional or malicious change.

Given the way autodetection appears intended to work, and the way the format libraries seem to be structured, that entire file ending business should be entirely unnecessary if the code is literally going to try every format until one returns success.